### PR TITLE
Added implementation from TextualButton

### LINKED ISSUE #412: Implement primary variants for Link
### Is your feature request related to a problem? Please describe ⚡️
Currently, we lack a way to bump the visual priority on the Link component. In cases where the button is too clunky. You want to use color to prioritise links.

### Describe the solution you'd like ✨
After some careful consideration we came up with a new pattern. Primary links. These will have a prop `variant="primary"` just like buttons. Which would render an alternate styling.

### PR DESCRIPTION
**Pull request by CarloPalinckx**
### This PR:

resolves #412 

**Breaking changes** 🔥
- None

**Backwards compatible additions** ✨
- The new component `TextualButton` has been added.

**Bugfixes/Changed internals** 🎈
- Moved tests for shared behaviour out of the `Button` test suite and moved them into a test suite for the button base.

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable).
- [x] Appropriate tests have been added for my functionality (check if not applicable).
- [x] A designer has seen and approved my changes (tag `@LuukHorsmans` or `@RianneSchaekens` for a design review when applicable).
- [x] I have tested my addition in all supported browsers and for responsiveness (Chrome, Firefox, Safari, Edge, IE11 and mobile browsers).
- [x] Appropriate documentation has been written (check if not applicable).

### CODE REVIEW

File: src/components/Button/base/test.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        const component = mount(
+            <MosTheme>
+                <Button title="button2 title" onClick={undefined} href="http://foo.bar" />
+            </MosTheme>,
+        );
+
+        expect(component.find('a').length).toBe(1);
+    });
+
+    it('should no-op on click without an onClick', () => {
```
Comment (hborghols) [inline src/components/Button/base/test.tsx]: This tests if it doesn't break, not if nothing happens.

**Comment by CarloPalinckx** [comment]
> Even though the checkbox suggests otherwise you didn't actually export the TextButton in index.ts

Yes I did 🤓 https://github.com/MyOnlineStore/bricks/pull/415/files#diff-92f135fa70f6d57d9f59b1397833a4caR58

**Comment by hborghols** [comment]
I'm talkin' bout `src/index.ts`

**Comment by CarloPalinckx** [comment]
Huh wait... I'm 100% sure I comitted that. Because the checkbox reminded me of it 😄. Coming right up...

**Comment by CarloPalinckx** [comment]
Screenshot 2019-07-25 at 10 13 26
🤦‍♂️

**Comment by hborghols** [comment]
😄 👍 👍

File: src/components/Button/base/style.tsx [hunk 2]
```diff
@@ -18,19 +19,8 @@ const StyledButton = styled.button<PropsType>`
...
-        left: 0;
-        top: 0;
-        right: 0;
-        bottom: 0;
-        z-index: -2;
-        opacity: 0;
-        transition: opacity 0.3s;
-    }
+    background: none;
+    border: none;
```
Comment (woutsluiter) [inline src/components/Button/base/style.tsx]: this one's already declared 🙃

**Comment by CarloPalinckx** [comment]
> this one's already declared 🙃

doh

**Pull request merged**